src/v/security/gssapi_authenticator.cc

```diff
@@ -419,4 +421,51 @@ void gssapi_authenticator::fail_impl(
     _state = state::failed;
 }
 
+acl_principal
+gssapi_authenticator::get_principal_from_name(std::string_view source_name) {
+    krb5::krb5_context krb5_ctx;
+    int krb5_rv = ::krb5_init_context(&krb5_ctx);
+    if (krb5_rv != 0) {
+        vlog(
+          seclog.error,
+          "Failed to initialize KRB5 instance for source_name mapping: {}",
+          krb5_rv);
+        return {};
+    }
+
+    const krb5::krb5_context_view krb5_ctx_view(&krb5_ctx);
+
+    krb5::krb5_default_realm default_realm(krb5_ctx_view);
+
+    krb5_rv = ::krb5_get_default_realm(
+      ::krb5_context{krb5_ctx}, &default_realm);
+
+    if (krb5_rv != 0) {
+        vlog(seclog.error, "Failed to obtain default realm: {}", krb5_rv);
```

Comment:
@BenPope do you know what this looks like when we log from the alien thread? like, what does it log as the shard id? wondering if there are other bad things lurking in there.

Reply:
I'm pretty sure logging is fine, it already has to be thread safe IIRC, there are sufficient barriers. I did specifically look at it when I put the other GSSAPI code on the `thread_,worker`.

Reply:
And there's no shard_id, and that's also ok with the logger.